tests/test_integration.py

```diff
@@ -378,3 +378,362 @@ def test(self, soco):
             'No difference between '
             'queues before and after removing the last item')
         assert len(new_queue) == len(old_queue) - 1
+
+
+class TestReorderSonosPlaylist(object):
+    """ Integration tests for Sonos Playlist Management. """
+    existing_playlists = None
+    pl_name = 'zSocoTestPlayList42'
+    test_pl = None
+
+    def _get_test_pl(self, soco):
+        '''return the DidlPlayListContainer for the pl_name'''
+        for spl in soco.get_sonos_playlists():
+            if spl.title == self.pl_name:
+                return spl
+
+    @pytest.yield_fixture(autouse=True, scope="class")
+    def restore_sonos_playlists(self, soco):
+        """ A fixture which cleans up after each sonos playlist test."""
+        if self.existing_playlists is None:
+            self.existing_playlists = soco.get_sonos_playlists()
+        if self.pl_name in [x.title for x in self.existing_playlists]:
+            msg = '%s is an existing playlist.' % self.pl_name
+            pytest.fail(msg)
```

Comment:
If this is meant for the unlikely case that someone might have a playlist by this name, then you could as python for a UUID. If it is just meant to check for missing cleanup after a failed test, then maybe the error message should say so.